src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ParseSecurityExpressionsPass.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Parses expressions used in the security access_control configuration to make sure they are dumped as SerializedParsedExpression
+ * This compiler pass must be registered after AddExpressionLanguageProvidersPass so custom functions can also be parsed.
+ *
+ * @author David Maicher <mail@dmaicher.de>
+ */
+class ParseSecurityExpressionsPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->has('security.expression_language')) {
+            return;
+        }
+
+        $expressionLanguage = $container->get('security.expression_language');
```

Comment:
`findDefinition()`

Reply:
No this should actually get the instantiated service :wink:

Reply:
Oh hm, I missed that. To be honest I am not really happy with having compiler passes that force the creation of the actual service during compilation. Can't we solve the problem by letting the serialization happening in a cache warmer where the service is actually available?

Reply:
I see your point. I'm also not 100% happy with this solution.

The only other option I see is to use a proper persisting cache by default for the expression language service. Currently it only uses an `ArrayAdapter`:

https://github.com/symfony/symfony/blob/master/src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php#L37

https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/SecurityBundle/Resources/config/security.xml#L82

On 3.3+ we could use the system cache pool for it?

And then we could add a `CacheWarmer` for it.

Or do you have another idea?

Reply:
Indeed, I had something like that in mind.

Reply:
I will have a look and maybe also prepare some benchmarks to see the impact of the change

Reply:
@xabbuh see https://github.com/symfony/symfony/pull/26263